docs/source/install/puppet.rst

```diff
@@ -0,0 +1,305 @@
+Puppet Module
+=============
+
+If you're ready to take complete control of your |st2| instances, then ``puppet-st2`` 
+is for you! It offers repeatable, configurable, and idempotent production-friendly
+|st2| installations.
+
+The source code for all the Puppet module is available as a GitHub repo: 
+`StackStorm/puppet-st2 <https://github.com/stackstorm/puppet-st2/>`_
+
+.. contents:: Contents
+   :local:
+
+---------------------------
+
+Supported Platforms
+-------------------
+
+The Puppet module supports the same platforms as manual installation, i.e.:
+
+* Ubuntu Trusty (14.04)
+* Ubuntu Xenial (16.04)
+* RHEL 6/CentOS 6
+* RHEL 7/CentOS 7
+
+The same system size :doc:`requirements </install/system_requirements>` also apply.
+
+Quick Start
+-----------
+
+The first step is installing Puppet, for this please consule the
+`Puppet Installation documentation <https://puppet.com/docs/puppet/latest/install_linux.html>`_
+
+To get started with a single node deployment, and default configuration settings, run these
+commands (expected to run as `root`):
+
+.. note::
+   
+  Puppet versions <= 3.x are no longer supported. Please utilize Puppet >= 4.
+
+.. code-block:: bash
+
+  # create a Puppetfile that defines our StackStorm module
+  echo -e "forge 'https://forgeapi.puppetlabs.com'\n mod 'stackstorm/st2'" > Puppetfile
```

Comment:
Instead of scripting, it would be nicer to extract this part to just highlight the resulting `Puppetfile` user needs to create via code block: 
```
https://forgeapi.puppetlabs.com
mod 'stackstorm/st2'
```

Hopefully would make it a bit more readable.